libr/bin/p/bin_dex.c

```diff
@@ -818,9 +831,15 @@ static RList* classes (RBinFile *arch) {
 	for (i = 0; i < bin->header.class_size; i++) {
 		ut64 class_addr = (ut64) bin->header.class_offset \
 			+ (sizeof (struct dex_class_t)*i);
-		// ETOOSLOW
-		r_buf_read_at (bin->b, class_addr, (ut8*)&entry,
-			sizeof (struct dex_class_t));
+		entry.class_id = r_read_le32 (bin->b->buf + class_addr + 0);
```

Comment:
Put those assignements in a separate function and maybe i would keep the old method in a #if block